data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,617 @@
+
+## Scopes and Environments
+
+Without oversimplifying, an OCaml program is a sequence of definitions or expressions. These items are said to be at the _top level_. An OCaml REPL, such as UTop, is called a toplevel because that's where typed expressions go.
+
+Execution evaluates each expression from top to bottom.
+
+At any time during evaluation, the _environment_ is the ordered sequence of available definitions.
+
+Here, the name `tau` is added into the toplevel environment.
+```ocaml
+# let tau = 6.28318;;
+val tau : float = 6.28318
+```
+
+The scope of `tau` is global. This name is available anywhere after the definition.
+
+Here, the global environment is unchanged:
+```ocaml
+# let pi = 3.14159 in 2. *. pi;;
+- : float = 6.28318
+```
+
+However, with respect to the expression `2. *. pi`, the environment is different because it contains the definition of `pi`. Local definitions create local environments.
+
+The scope of a definition is the set of environments where it is reachable.
+
+Although OCaml is an expression-oriented language, it is not entirely free of statements. The `let` construct is a statement that may change the state of the environment by adding a name-value binding. In some sense, the `let` is the only statement OCaml has. Note these expressions at the toplevel also fall into that category because they are equivalent to `let _ =` definitions.
+```ocaml
+# (1.0 +. sqrt 5.0) /. 2.0;;
+- : float = 1.6180339887498949
+
+# let _ = (1.0 +. sqrt 5.0) /. 2.0;;
+- : float = 1.6180339887498949
+```
+
+With respect to the environment, there are no means to:
+- List its contents
+- Remove a definition
+- Reset it to an earlier state
+
+### Same-Level Shadowing
+
+*Shadowing* can be a difficult concept to grasp. As you learned in the [Tour of OCaml](/docs/tour-of-ocaml), bindings are immutable. Once you create a name and give it a value (definition), it does not change. They are bound together. That said, a name can be defined again
+
+You can write two definitions using the same identifier in sequence:
+```ocaml
+# let c = 2 * 3;;
+val c : int = 6
+
+# let d = c * 7;;
+val d : int = 42
+
+# let c = 7 * 7;;
+val c : int = 49
+
+# d;;
+- : int = 42
+
+# c;;
+- : int = 49
+```
+
+In this example, `c` is defined twice. The key thing to understand is that the name `c` is *not updated*. It looks as if the value of the first `c` has changed, but it hasn't. When the second `c` is defined, the first one becomes unreachable, but it still remains in the environment. This means anything written after the second definition uses its value, but anything written *before* the second definition, still uses the first definition, even if its called after. 
+
+In the above example, `d` remains `42` even though `c` was defined again as `7 * 7`, or `49`. Since `d` was defined before the value of `c` was changed, it still uses the first `c` definition. This is known as a *closure,* which is [explored later in this document](#closures).
```

Comment:
```suggestion

```
I'm removing this at least because it's not completely accurate, and Thibaut said this is the same behavior in other languages when a variable is redefined.